### ISSUE TITLE
A Exception When I stop worker : Unsupported operand types

### ISSUE THREAD

**Issue by davidtall** [mjphaynes/php-resque#64]
This is my cofig.yml file :
```
redis:
     scheme: tcp         # The connection scheme
     host: 10.27.168.117
     port: 6379          # The port to connect to
     password: xxx          # The port to connect to
     namespace: resque   # The key prefix in Redis
     rw_timeout: 60      # Redis read/write timeout
     phpiredis: false    # Should use phpiredis php extension (must be installed)


default:
     expiry_time: 604800 # Default expiry time of Redis keys
     #verbose:           # quiet=0, normal=1, verbose=2, very_verbose=3, debug=4
     #jobs:
         #queue: default  # Defalt queue to add a job to
         #delay: false    # Default delay to add to jobs false=no delay
     workers:
         queue: "*"      # Default queue to watch, *=all queues
         blocking: true  # Use Redis blocking
         interval: 1    # Worker interval (seconds)
         timeout: 60     # Job timeout (seconds)
         memory: 128     # Max worker memory
         #pid: 20088     # Max worker memory
```
First ,I start two woker on one machine,and add 1000 jobs into queue:
```
   for($i=0;$i<1000;$i++){
            \Resque::push('Open\\Job\\TestJob', array(), "shop" . mt_rand(1, 100));
        }
```
not until all jobs was processed,I run command `resque worker:stop` to stop woker.
One of the workers throw a exception :`Unsupported operand types` in src/Resque/Worker.php(787)
then I debug it,
https://github.com/mjphaynes/php-resque/blob/8ec37dbcde8fbf4088efd2385cc015f765ec16da/src/Resque/Worker.php#L765
I find that `$found` return a empty array,So I log here like that:
```
 list($jobs, $found) = $res=$this->redis->exec();
 if(is_array($found)){
                $this->log(var_export($res,true), Logger::NOTICE);
            }
```
then I retry the case many times (It does not appear every times), the log file record :
```
[2017-09-03 14:57:43] resque.INFO: Done job shop54:Open\Job\TestJob#21075b4615e468768e61fd() in 8.2ms [] []
[2017-09-03 14:57:43] resque.INFO: Done job shop54:Open\Job\TestJob#21075b4615e46876a27ce5() in 7.4ms [] []
[2017-09-03 14:57:43] resque.DEBUG: QUIT received; shutdown worker [] []
[2017-09-03 14:57:43] resque.NOTICE: array (   0 =>    Predis\Response\Status::__set_state(array(      'payload' => 'OK',   )),   1 =>    array (   ),   2 => 0, ) [] []
```
I don't know why !
https://github.com/mjphaynes/php-resque/blob/8ec37dbcde8fbf4088efd2385cc015f765ec16da/src/Resque/Worker.php#L760-L763

Why does It return three  elements after `$this->redis->exec()`?
@francislavoie @xelan

**Comment by francislavoie** [comment]
So my understanding of the `multi()` API is that is batches together multiple commands in a row, where `exec()` will return the result of the set of commands in an array.

https://redis.io/commands/multi
https://redis.io/commands/zrangebyscore
https://redis.io/commands/zremrangebyscore
https://redis.io/commands/exec

So I think the first element is just a status element for the `multi()` batch that's always returned, then the next two are the `$jobs` array and the `$found` number of removals by `zremrangebyscore`.

Just a guess here, but that code looks offset, i.e. it's not grabbing the right variables as output.

Maybe try `list($status, $jobs, $found)` so that the first argument, which is the status of the transaction (see https://redis.io/topics/transactions#usage) gets read in first.

I have no idea if it'll work better, but those are just my thoughts based on what I'm seeing. I never tried running the stop command because I don't need it personally.

**Comment by davidtall** [comment]
But I don't add any delayed jobs into queue

**Comment by francislavoie** [comment]
It gets called from https://github.com/mjphaynes/php-resque/blob/master/src/Resque/Worker.php#L267 on every loop.

Side-notes: I still recommend you increase you interval setting to something much higher than 1 second. At 1 second, your script will wake back up every second needlessly. You should let it sit quietly for longer to prevent more CPU usage. The interval means how long it will let the `BLPOP` command wait on each loop. It's safe to let it wait a long time, because you'll immediately get a job popped back when one's available. That's the purpose of the blocking aspect of that command. In addition, you don't have `phpiredis` turned on, you'll probably get much better redis performance by installing it and turning it on.

**Comment by davidtall** [comment]
Oh thanks ~ I have another question:
https://github.com/mjphaynes/php-resque/blob/8ec37dbcde8fbf4088efd2385cc015f765ec16da/src/Resque/Worker.php#L320
If I use `pcntl_wait($status,WNOHANG);` instead of `pcntl_wait($status);` ,will the job processed quickly?
Does it means that all child process can run at the same time? What wrong thing will happen if I set that?

**Comment by francislavoie** [comment]
No, that would be bad. It would mean that the worker will not wait for the child to finish the job, so it'll not be able to guarantee getting back the response and not be able to report the status of the job correctly. It won't speed up job processing, it'll just make your worker fork jobs constantly without actually reporting success. Like I said before, just run more jobs if you want more parallelism. You could use something like systemd to manage the processes and it'll keep multiple workers alive. I use docker and I use the `scale` option to run multiple workers in containers.

**Comment by francislavoie** [comment]
Did you test the `list($status, $jobs, $found)` change? Did that fix anything? If so, would you mind making another PR for it?

**Comment by davidtall** [comment]
It returns 2 element under normal conditions,but some times return 3 element.
So ,I  add ` if(!is_array($res) || count($res) != 2) return;`  to resolve the problem.

**Comment by xelan** [comment]
Bug reproducable on Debian 9.1, PHP 7.1.9-1+0~20170902060745.8+stretch~1.gbpebe5d6

**Comment by xelan** [comment]
Hi @davidtall and @francislavoie!

I'm closing this issue in favor of the PR #66 so we don't have two places to discuss the same issue.

Best regards
Andreas
